Replace gdp with Go 1.6+ built-in vendoring, update all deps to latest

snwight opened the pull request:
Replace gdp with Go 1.6+ built-in vendoring. Updated all deps to latest versions, vendored them all so we have a snapshot of today's state. I used govendor to init and sync,  which is a sledgehammer approach and pulls in every last thing, which is what we probably want for this docker env. I made a single change to the source, in cmd/hologram-authorize/main.go to adapt for a recent change to gopass package. This builds and all tests - such as they are - pass under Go 1.7.1

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AdRoll/hologram?pullRequest=82) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AdRoll/hologram?pullRequest=82) before we can accept your contribution.<br/><hr/>**Steve Wight** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

snwight commented:
Low hanging fruit, part deux... cleaned up header comments for packages so the 'Overview' section isn't filled with license info, added godoc generation target to buildscripts (**generate_docs.sh**) which serves up HTML godocs to localhost:6060 as well as static type analysis of source (so the source refs have live ref links, useful for navigation in the absence of a Go-sensitive editor env)

copumpkin commented on .idea/libraries/GOPATH__hologram_.xml:
```diff
@@ -0,0 +1,15 @@
+      <root url="file://$USER_HOME$/src/go/src/github.com" />
+    </CLASSES>
+    <SOURCES>
+      <root url="file://$USER_HOME$/src/go/src/golang.org" />
```
These seem rather user-specific, right?

snwight replied:
just making me crazy - using jetbrains and i keep accidentally committing this crap... hopefully removed in next commit. thanks!

snwight commented:
thanks for the observation, @copumpkin - i'm a recent convert to jetbrains IDE (from emacs) ... adding `.idea/` to .gitignore now
